s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/config/S3ClientFactory.java

```diff
@@ -16,45 +16,50 @@
 
 package io.aiven.kafka.connect.s3.source.config;
 
+import java.net.URI;
+import java.time.Duration;
 import java.util.Objects;
+import java.util.Random;
 
 import io.aiven.kafka.connect.iam.AwsCredentialProviderFactory;
 
-import com.amazonaws.PredefinedClientConfigurations;
-import com.amazonaws.client.builder.AwsClientBuilder;
-import com.amazonaws.retry.PredefinedBackoffStrategies;
-import com.amazonaws.retry.PredefinedRetryPolicies;
-import com.amazonaws.retry.RetryPolicy;
-import com.amazonaws.services.s3.AmazonS3;
-import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.retry.RetryMode;
+import software.amazon.awssdk.retries.api.internal.backoff.ExponentialDelayWithJitter;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3Configuration;
 
 public class S3ClientFactory {
 
     private final AwsCredentialProviderFactory credentialFactory = new AwsCredentialProviderFactory();
 
-    public AmazonS3 createAmazonS3Client(final S3SourceConfig config) {
-        final var awsEndpointConfig = newEndpointConfiguration(config);
-        final var clientConfig = PredefinedClientConfigurations.defaultConfig()
-                .withRetryPolicy(new RetryPolicy(PredefinedRetryPolicies.DEFAULT_RETRY_CONDITION,
-                        new PredefinedBackoffStrategies.FullJitterBackoffStrategy(
-                                Math.toIntExact(config.getS3RetryBackoffDelayMs()),
-                                Math.toIntExact(config.getS3RetryBackoffMaxDelayMs())),
-                        config.getS3RetryBackoffMaxRetries(), false));
```

Comment:
I think you dropped this maxRetries!  You can add it later to the `r -> r .backoffStrategy(backoffStrategy).maxAttempts(config.getS3RetryBackoffMaxRetries()`

Reply:
updated and also updated the new method names conventions.